dashboard/modules/runtime_env/runtime_env_agent.py

```diff
@@ -72,6 +78,11 @@ def run_setup_with_logger():
                 # setup hooks ran for arbitrary job in arbitrary threads.
                 with using_thread_local_logger(per_job_logger):
                     env_context = self._setup(runtime_env, session_dir)
+                    if "uris" in runtime_env:
+                        working_dir = runtime_env_pkg.ensure_runtime_env_setup(
```

Comment:
We use `ensure_runtime_env_setup ` directly here. In long term, we should make this function pluggable like `runtime_env_setup_hook `.  We can also optimize this in future PRs because the runtime env pluggable framework should also be optimized in future.

Reply:
Yes absolutely!